reverse_replication/launcher.go

```diff
@@ -0,0 +1,273 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	dataflow "cloud.google.com/go/dataflow/apiv1beta3"
+	"cloud.google.com/go/dataflow/apiv1beta3/dataflowpb"
+	"cloud.google.com/go/pubsub"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/storage"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+var (
+	projectId            string
+	dataflowRegion       string
+	jobNamePrefix        string
+	changeStreamName     string
+	instanceId           string
+	dbName               string
+	metadataInstance     string
+	metadataDatabase     string
+	pubSubDataTopicId    string
+	pubSubEndpoint       string
+	sourceShardsFilePath string
+	sessionFilePath      string
+)
+
+func setupGlobalFlags() {
+	flag.StringVar(&projectId, "projectId", "", "projectId")
+	flag.StringVar(&dataflowRegion, "dataflowRegion", "", "region for dataflow jobs")
+	flag.StringVar(&jobNamePrefix, "jobNamePrefix", "reverse-rep", "job name prefix for the dataflow jobs")
+	flag.StringVar(&changeStreamName, "changeStreamName", "", "change stream name")
+	flag.StringVar(&instanceId, "instanceId", "", "spanner instance id")
+	flag.StringVar(&dbName, "dbName", "", "spanner database name")
+	flag.StringVar(&metadataInstance, "metadataInstance", "", "spanner instance name to store changestream metadata")
+	flag.StringVar(&metadataDatabase, "metadataDatabase", "", "spanner database name to store changestream metadata")
```

Comment:
We could give a default database name like change_stream_metadata

Reply:
Currently, it does not auto create a db if one doesnt exist. Will add that feature and default to this.